test/AutoDiff/Sema/transpose_attr_type_checking.swift

```diff
@@ -472,6 +473,7 @@ extension Float {
 
 // Test non-`func` original declarations.
 
+// expected-note @+1 {{candidate initializer does not have type equal to or less constrained than '<T where T : Differentiable, T == T.TangentVector> (Struct<T>) -> (Float) -> Struct<T>'}}
 struct Struct<T> {}
```

Comment:
Note: this diagnostic note appears for a compiler-synthesized struct memberwise initializer, which isn't explicit in source code. I think the note is pretty clear, it's emitted on the struct declaration and includes `candidate *initializer*`.